lib/spack/spack/binary_distribution.py

```diff
@@ -698,7 +698,7 @@ def write_buildinfo_file(spec, workdir, rel=False):
     prefix_to_hash = dict()
     prefix_to_hash[str(spec.package.prefix)] = spec.dag_hash()
     deps = spack.build_environment.get_rpath_deps(spec.package)
-    for d in deps:
+    for d in deps + spec.dependencies():
```

Comment:
I think this should be `spec.dependencies(type="run")` since build dependencies shouldn't be affected when installing from binaries. Also, it might be better to give this set of specs a name and extract a function.

Reply:
Is there any harm in including "build" dependencies? I can imagine that some package authors do not realize when a "build" dependency actually also is a "run" dependency because spack writes the full path to the dependency's executable into some run script of the dependent.

Reply:
It would be wrong from a modeling perspective. If we realize that a "build" dependency is also "run" we should modify the package, since the model affect contexts beyond just relocation.

Reply:
OK, I'll change it.